pandas/core/categorical.py

```diff
@@ -2068,7 +2068,7 @@ def _delegate_method(self, name, *args, **kwargs):
         method = getattr(self.categorical, name)
         res = method(*args, **kwargs)
         if res is not None:
-            return Series(res, index=self.index)
+            return Series(res, index=self.index, name=self.name)
```

Comment:
Ah, I misread the earlier. `self` is the instance of `CategoricalAccessor`, not the `Series` like I expected. So this approach won't work.

Reply:
so, what do you think will work for this scenario?

Reply:
I think if you modify https://github.com/pandas-dev/pandas/blob/f11bbf2f505d81900cc83ce387a6a1b1d2a2f866/pandas/core/categorical.py#L2078 to get `data.name`. and then modify `CategoricalAccessor.__init__` to accept a name. `data` may not have a name, so maybe `getattr(data, 'name', None)`.

Reply:
so the signature should follow like we do in ``pandas.core.indexes.accessors``, e.g. ``(data, index, name=None)``